diesel/src/pg/types/date_and_time/mod.rs

```diff
@@ -143,17 +143,18 @@ impl ToSql<sql_types::Interval, Pg> for PgInterval {
 }
 
 impl FromSql<sql_types::Interval, Pg> for PgInterval {
-    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
-        let bytes = not_none!(bytes);
+    fn from_sql(value: Option<PgValue>) -> deserialize::Result<Self> {
+        let value = not_none!(value);
+        let bytes = value.bytes();
         Ok(PgInterval {
             microseconds: try!(FromSql::<sql_types::BigInt, Pg>::from_sql(Some(
-                &bytes[..8]
+                PgValue::with_oid(bytes[..8].as_ptr() as *mut u8, 8, 0) // TODO Find OID
```

Comment:
We need to find some way to get the oid from the `Pg: HasSqlType<BigInt>` implementation. That basically means we need to find some way to construct a fake `PgMetadataLookup` struct here. (The implementation will not use it internally because the oid is static for `BigInt`)